ui/app/services/path-help.js

```diff
@@ -34,119 +36,159 @@ export default Service.extend({
-    //if we have a factory, we need to take the existing model into account
+    // if we have a factory, we need to take the existing model into account
     if (modelFactory) {
       debug(`Model factory found for ${modelType}`);
       newModel = modelFactory.class;
       const modelProto = newModel.proto();
       if (newModel.merged || modelProto.useOpenAPI !== true) {
         return resolve();
       }
+
       helpUrl = modelProto.getHelpUrl(backend);
       return this.registerNewModelWithProps(helpUrl, backend, newModel, modelName);
     } else {
       debug(`Creating new Model for ${modelType}`);
       newModel = DS.Model.extend({});
-      //use paths to dynamically create our openapi help url
-      //if we have a brand new model
-      return this.getPaths(apiPath, backend, itemType).then(paths => {
+    }
+
+    // we don't have an apiPath for dynamic secrets
+    // and we don't need paths for them yet
+    if (!apiPath) {
+      helpUrl = newModel.proto().getHelpUrl(backend);
+      return this.registerNewModelWithProps(helpUrl, backend, newModel, modelName);
+    }
+
+    // use paths to dynamically create our openapi help url
+    // if we have a brand new model
+    return this.getPaths(apiPath, backend, itemType)
+      .then(pathInfo => {
         const adapterFactory = owner.factoryFor(`adapter:${modelType}`);
-        //if we have an adapter already use that, otherwise create one
+        // if we have an adapter already use that, otherwise create one
         if (!adapterFactory) {
           debug(`Creating new adapter for ${modelType}`);
-          const adapter = this.getNewAdapter(paths, itemType);
+          const adapter = this.getNewAdapter(pathInfo, itemType);
           owner.register(`adapter:${modelType}`, adapter);
         }
-        //if we have an item we want the create info for that itemType
-        let path;
-        if (itemType) {
-          const createPath = paths.create.find(path => path.path.includes(itemType));
-          path = createPath.path;
-          path = path.slice(0, path.indexOf('{') - 1) + '/example';
-        } else {
-          //we need the mount config
-          path = paths.configPath[0].path;
+        let path, paths;
+        // if we have an item we want the create info for that itemType
+        paths = itemType ? this.filterPathsByItemType(pathInfo, itemType) : pathInfo.paths;
+        const createPath = paths.find(path => path.operations.includes('post') && path.action !== 'Delete');
+        path = createPath.path;
+        path = path.includes('{') ? path.slice(0, path.indexOf('{') - 1) + '/example' : path;
+        if (!path) {
+          // TODO: we don't know if path will ever be falsey
+          // if it is never falsey we can remove this.
+          return reject();
         }
-        helpUrl = `/v1/${apiPath}${path.slice(1)}?help=true`;
+
+        helpUrl = `/v1/${apiPath}${path.slice(1)}?help=true` || newModel.proto().getHelpUrl(backend);
```

Comment:
this is how we construct the URL for making the request to OpenAPI. if the `?help-true` suffix ever changes for some reason you'll need to update it here.